### PR TITLE
Fix dns for azure module

### PR DESCRIPTION
**Pull request by tei-k**
# Description
https://scalar-labs.atlassian.net/browse/DLT-5652

# Done
- Add suffix `count.index` to dns.
- Fix to can be create multi bastion.

### CODE REVIEW

File: modules/azure/monitor/main.tf [hunk 1]
```diff
@@ -90,26 +90,48 @@ module "monitor_provision" {
   internal_domain               = local.internal_domain
 }
 
-resource "azurerm_private_dns_a_record" "monitor-dns" {
+resource "azurerm_private_dns_a_record" "monitor" {
```
Comment (feeblefakie) [inline modules/azure/monitor/main.tf:93]: How about `monitor-cluster-dns` for monitor.xyz.com and `monitor-host-dns` for monitor-1.xyz.com ?

BTW, all the dns variables are not very consistent. Some uses `-` and others use `_` instead, also some has `a` but the others don't.

Comment (feeblefakie) [reply modules/azure/monitor/main.tf:93]: @tei-k Please take a look at the comment !

Comment (tei-k) [reply modules/azure/monitor/main.tf]: Thanks comment! Changed below.

- Fix resource name to  `monitor_cluster_dns` and `monitor_host_dns`.
- Fix `-` to `_` in resoruce_name.
- Delete prometheus dns name. (doesn't seem necessary)

Comment (ymorimo) [reply modules/azure/monitor/main.tf]: > Delete prometheus dns name. (doesn't seem necessary)

There is a DNS record resource for prometheus on AWS side too. Let's remove it later.

**Pull request merged**